src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1409,6 +1424,102 @@ def disable_features(cmd, client, resource_group_name, cluster_name, features, k
+        load_kube_config(kube_config, kube_context)
+        configuration = kube_client.Configuration()
+        try:
+            validate_release_namespace(client, cluster_name, resource_group_name, configuration, kube_config, kube_context)
+        except Exception as e:
+            logger.error(str(e))
+            tr_logger.error(str(e))
+
+        kapi_instance = kube_client.CoreV1Api(kube_client.ApiClient(configuration))
+        try:
+            pod_list = kapi_instance.list_namespaced_pod(consts.Arc_Namespace)
+            pods_count = 0
+            for pod in pod_list.items:
+                pods_count += 1
+                tr_logger.warning("Pod {} is in {} state. Reason: {}. Container statuses: {}".format(pod.metadata.name, pod.status.phase, pod.status.reason, pod.status.container_statuses))
+
+            if pods_count == 0:
+                tr_logger.warning("No pods found in azure-arc namespace.")
+
+        except Exception as ex:
+            tr_logger.error("Error occured while fetching pod's statues : {}".format(str(ex)))
+
+        cert_secret = utils.get_kubernetes_secret(kapi_instance, consts.Arc_Namespace, consts.AZURE_IDENTITY_CERTIFICATE_SECRET, custom_logger=tr_logger)
+        if (not cert_secret) or (not hasattr(cert_secret, 'data')) or (consts.AZURE_IDENTITY_CERTIFICATE_SECRET not in cert_secret.data):
+            tr_logger.error("{} secret is not present on the kubernetes cluster.".format(consts.AZURE_IDENTITY_CERTIFICATE_SECRET))
+            logger.error("{} secret is not present on the kubernetes cluster.".format(consts.AZURE_IDENTITY_CERTIFICATE_SECRET))
+
+        try:
+            cc_object = json.loads(connected_cluster.response.content)
+            raw_exp_time = cc_object.get("properties").get("managedIdentityCertificateExpirationTime")
+            if raw_exp_time:
+                cert_expirn_time = datetime.strptime(raw_exp_time, consts.ISO_861_Time_format).replace(tzinfo=timezone.utc)
+                current_time = datetime.now(timezone.utc)
+                if cert_expirn_time != datetime.min and cert_expirn_time < current_time:
+                    logger.error("MSI certificate on the cluster has expired.")
+        except Exception as ex:
+            pass
+
+        storage_account_name, sas_token, readonly_sas_token = utils.setup_validate_storage_account(cmd.cli_ctx, storage_account, sas_token, resource_group_name)
+        if storage_account_name:  # When validated the storage account
+            utils.try_upload_log_file(cluster_name, storage_account_name, sas_token, troubleshoot_log_path)
+            utils.try_archive_log_file(troubleshoot_log_path, output_file)
+            # token_in_storage_account_url = readonly_sas_token if readonly_sas_token is not None else sas_token.
+            utils.collect_periscope_logs(resource_group_name, cluster_name, storage_account_name, sas_token, readonly_sas_token, kube_context, kube_config)
```

Comment:
One imp concern I have regarding periscope:
For cases when the cluster is behind proxy, periscope will not be able to push pod logs to the storage which defeats the whole purpose of troubleshoot. Somehow, we should find a way to get pod logs on the local machine. Since we have seen many issues due to proxy in the past.

Reply:
That must be handled by the aks-periscope internally I guess, because aks-periscope when collecting logs for aks cluster behind the proxy, should have implemented that. 
@sophsoph321 Can you confirm back on this once?

Reply:
Yes, part of the implementation for getting the pod logs on the local machine must be handled by aks-periscope. Currently, this work will not be in scope for the first release of this troubleshoot cli command, but will be done in the next release.